Hi @JulesL2 thank you for the contribution! 

I'm getting an error when I try and run your code. Looks like an issue with the updated regex. I'll take a closer look when I get a minute. 

```
2024.07.02 08:35:32 - [info] - starting Blue Iris Exporter with the following:
                Log Path: C:\Users\Garret\Desktop\github\log
                Metric Path: /metrics
                Port: :2112
panic: regexp: Compile(`(?P<folder>[^\s\\]*)(\s*Delete:).+(\[((\s(?P<sizeused>[\d\.]+)(?P<sizeunit>\w*)\/(?P<sizelimit>[\d\.]+)(?P<sizelimitunit>\w+)),\s((?P<diskfree>[\d\.]+)(?P<freeunit>\w+))\sfree\])`): error parsing regexp: missing closing ): `(?P<folder>[^\s\\]*)(\s*Delete:).+(\[((\s(?P<sizeused>[\d\.]+)(?P<sizeunit>\w*)\/(?P<sizelimit>[\d\.]+)(?P<sizelimitunit>\w+)),\s((?P<diskfree>[\d\.]+)(?P<freeunit>\w+))\sfree\])`

goroutine 20 [running]:
regexp.MustCompile({0x6de796, 0xb2})
        C:/Program Files/Go/src/regexp/regexp.go:317 +0xbb
github.com/wymangr/blueiris_exporter/blueiris.findObject({0xc000033d40, 0x60})
        C:/Users/User/Desktop/github/blueiris_exporter/blueiris/blueirisMetrics.go:418 +0x1009
github.com/wymangr/blueiris_exporter/blueiris.BlueIris(0xc00007e240?, {0xc00021ee00, 0x2, {0x6c1bd8, 0xb}, 0x1, {0xc000248000, 0x15, 0x15}, 0x6fb8b0, ...}, ...)
        C:/Users/User/Desktop/github/blueiris_exporter/blueiris/blueirisMetrics.go:102 +0x5c5
main.CollectMetrics(0x0?, 0x0?, {0xc00021ee00, 0x2, {0x6c1bd8, 0xb}, 0x1, {0xc000248000, 0x15, 0x15}, ...}, ...)
        C:/Users/User/Desktop/github/blueiris_exporter/metrics.go:112 +0x34b
created by main.(*ExporterBlueIris).Collect
        C:/Users/User/Desktop/github/blueiris_exporter/blueiris_exporter.go:38 +0x285
exit status 2
```

--- a/blueiris/blueirisMetrics.go
+++ b/blueiris/blueirisMetrics.go
@@ -45,6 +45,7 @@ var (
 	errorMetrics        map[string]float64
 	warningMetrics      map[string]float64
 	parseErrors         map[string]float64
+	profileCount        map[string]float64
 )
 
 func BlueIris(ch chan<- prometheus.Metric, m common.MetricInfo, SecMet []common.MetricInfo, logpath string) {
@@ -55,6 +56,7 @@ func BlueIris(ch chan<- prometheus.Metric, m common.MetricInfo, SecMet []common.
 	warningMetrics = make(map[string]float64)
 	triggerCount = make(map[string]float64)
 	pushCount = make(map[string]float64)
+	profileCount = make(map[string]float64)
 	errorMetricsTotal = 0
 	warningMetricsTotal = 0
 	parseErrorsTotal = 0
@@ -199,7 +201,10 @@ func BlueIris(ch chan<- prometheus.Metric, m common.MetricInfo, SecMet []common.
 				detail := details[2]
 				ch <- prometheus.MustNewConstMetric(sm.Desc, sm.Type, v, camera, status, detail)
 			}
-
+		case "profile":
+			for f, v := range profileCount {
+				ch <- prometheus.MustNewConstMetric(sm.Desc, sm.Type, v, f)
+			}
 		case "parse_errors":
 			if len(parseErrors) == 0 {
 				ch <- prometheus.MustNewConstMetric(sm.Desc, sm.Type, 0, "")
@@ -289,7 +294,7 @@ func convertBytes(s string, unit string) (f float64, err error) {
 		convertedFloat, errConv = convertStrFloat(s)
 		bytefl = convertedFloat * 1024 * 1024 * 1024 * 1024
 	} else {
-		common.BIlogger(fmt.Sprintf("s %s , Unit %s", s,unit), "console")
+		common.BIlogger(fmt.Sprintf("s %s , Unit %s", s, unit), "console")
 		errConv = errors.New("unable to determine sting type (B, K, M, G, T)")
 	}
 	return bytefl, errConv
@@ -392,11 +397,25 @@ func findObject(line string) (match []string, r *regexp.Regexp, matchType string
 			}
 			camerastatus[camera]["detail"] = status
 		}
+	} else if strings.Contains(line, "Current profile:") {
+		r := regexp.MustCompile(`(App)(\s*Current profile:\s)(?P<profile>.+)`)
+		match := r.FindStringSubmatch(line)
+		if len(match) == 0 {
+			parseErrors = appendCounterMap(parseErrors, line)
+			parseErrorsTotal++
+		} else {
+			profileMatch := r.SubexpIndex("profile")
+			profile := match[profileMatch]
+			for f, _ := range profileCount {
+				profileCount[f] = 0
+			}
+			profileCount[profile] = 1
+		}
 	} else if strings.Contains(line, "Delete: ") && strings.HasPrefix(line, "0 ") {
 		r := regexp.MustCompile(`(?P<folder>[^\s\\]*)(\s*Delete:).+(\[(((?P<hoursused>[0-9]*)\/(?P<hourstotal>[0-9]*))\shrs,(\s(?P<sizeused>[\d\.]+)(?P<sizeunit>\w*)\/(?P<sizelimit>[\d\.]+)(?P<sizelimitunit>\w+)),\s((?P<diskfree>[\d\.]+)(?P<freeunit>\w+))\sfree)\])`)
 		match := r.FindStringSubmatch(line)
 		if len(match) == 0 {
-			r1 := regexp.MustCompile(`(?P<folder>[^\s\\]*)(\s*Delete:).+(\[((\s(?P<sizeused>[\d\.]+)(?P<sizeunit>\w*)\/(?P<sizelimit>[\d\.]+)(?P<sizelimitunit>\w+)),\s((?P<diskfree>[\d\.]+)(?P<freeunit>\w+))\sfree\])`)
+			r1 := regexp.MustCompile(`(?P<folder>[^\s\\]*)(\s*Delete:).+\[(((?P<sizeused>[\d\.]+)(?P<sizeunit>\w*)\/(?P<sizelimit>[\d\.]+)(?P<sizelimitunit>\w+)),\s((?P<diskfree>[\d\.]+)(?P<freeunit>\w+))\sfree.+)`)
 			match1 := r1.FindStringSubmatch(line)
 			if len(match1) == 0 {
 				parseErrors = appendCounterMap(parseErrors, line)
@@ -412,20 +431,20 @@ func findObject(line string) (match []string, r *regexp.Regexp, matchType string
 			freeunitMatch1 := r1.SubexpIndex("freeunit")
 
 			folder1 := match1[folderMatch1]
-			
-			sizelimit1, err := convertBytes(match1[sizelimitMatch1],match1[sizelimitunitMatch1])
+
+			sizelimit1, err := convertBytes(match1[sizelimitMatch1], match1[sizelimitunitMatch1])
 			if err != nil {
 				common.BIlogger(fmt.Sprintf("Unable to get convert sizelimit Error: \n%v", err), "console")
 				return nil, nil, ""
 			}
-			diskfree1, err := convertBytes(match1[diskfreeMatch1],match1[freeunitMatch1])
+			diskfree1, err := convertBytes(match1[diskfreeMatch1], match1[freeunitMatch1])
 			if err != nil {
 				common.BIlogger(fmt.Sprintf("Unable to get convert diskfree Error: \n%v", err), "console")
 				return nil, nil, ""
 			}
-			
-			if strings.Compare(match1[sizeunitMatch1],"") == 0 {
-				sizeused1, err := convertBytes(match[sizeusedMatch1],match1[sizelimitunitMatch1])
+
+			if strings.Compare(match1[sizeunitMatch1], "") == 0 {
+				sizeused1, err := convertBytes(match1[sizeusedMatch1], match1[sizelimitunitMatch1])
 				if err != nil {
 					common.BIlogger(fmt.Sprintf("Unable to get convert sizeused Error: \n%v", err), "console")
 					return nil, nil, ""
@@ -437,7 +456,7 @@ func findObject(line string) (match []string, r *regexp.Regexp, matchType string
 				diskStats[folder1]["diskfree"] = diskfree1
 				diskStats[folder1]["sizePercent"] = sizePercent1
 			} else {
-				sizeused1, err := convertBytes(match[sizeusedMatch1], match1[sizeunitMatch1])
+				sizeused1, err := convertBytes(match1[sizeusedMatch1], match1[sizeunitMatch1])
 				if err != nil {
 					common.BIlogger(fmt.Sprintf("Unable to get convert sizeused Error: \n%v", err), "console")
 					return nil, nil, ""
@@ -448,7 +467,7 @@ func findObject(line string) (match []string, r *regexp.Regexp, matchType string
 				}
 				diskStats[folder1]["diskfree"] = diskfree1
 				diskStats[folder1]["sizePercent"] = sizePercent1
-			}		
+			}
 
 		} else {
 			folderMatch := r.SubexpIndex("folder")
@@ -472,12 +491,12 @@ func findObject(line string) (match []string, r *regexp.Regexp, matchType string
 				common.BIlogger(fmt.Sprintf("Unable to convert hourstotal string to float: \n%v", hourstotal), "console")
 				return nil, nil, ""
 			}
-			sizelimit, err := convertBytes(match[sizelimitMatch],match[sizelimitunitMatch])
+			sizelimit, err := convertBytes(match[sizelimitMatch], match[sizelimitunitMatch])
 			if err != nil {
 				common.BIlogger(fmt.Sprintf("Unable to get convert sizelimit Error: \n%v", err), "console")
 				return nil, nil, ""
 			}
-			diskfree, err := convertBytes(match[diskfreeMatch],match[freeunitMatch])
+			diskfree, err := convertBytes(match[diskfreeMatch], match[freeunitMatch])
 			if err != nil {
 				common.BIlogger(fmt.Sprintf("Unable to get convert diskfree Error: \n%v", err), "console")
 				return nil, nil, ""
@@ -485,9 +504,8 @@ func findObject(line string) (match []string, r *regexp.Regexp, matchType string
 
 			hourPercent := (hoursused / hourstotal) * 100
 
-
-			if strings.Compare(match[sizeunitMatch],"") == 0 {
-				sizeused, err := convertBytes(match[sizeusedMatch],match[sizelimitunitMatch])
+			if strings.Compare(match[sizeunitMatch], "") == 0 {
+				sizeused, err := convertBytes(match[sizeusedMatch], match[sizelimitunitMatch])
 				if err != nil {
 					common.BIlogger(fmt.Sprintf("Unable to get convert sizeused Error: \n%v", err), "console")
 					return nil, nil, ""
@@ -497,7 +515,7 @@ func findObject(line string) (match []string, r *regexp.Regexp, matchType string
 				if _, ok := diskStats[folder]; !ok {
 					diskStats[folder] = make(map[string]float64)
 				}
-	
+
 				diskStats[folder]["diskfree"] = diskfree
 				diskStats[folder]["hourPercent"] = hourPercent
 				diskStats[folder]["sizePercent"] = sizePercent
@@ -512,7 +530,7 @@ func findObject(line string) (match []string, r *regexp.Regexp, matchType string
 				if _, ok := diskStats[folder]; !ok {
 					diskStats[folder] = make(map[string]float64)
 				}
-	
+
 				diskStats[folder]["diskfree"] = diskfree
 				diskStats[folder]["hourPercent"] = hourPercent
 				diskStats[folder]["sizePercent"] = sizePercent

--- a/metrics.go
+++ b/metrics.go
@@ -21,7 +21,7 @@ var (
 	namespace string = "blueiris"
 
 	blueIrisServerMetrics = metrics{
-		1:  newMetric("ai_duration", "Duration of Blue Iris AI analysis", prometheus.GaugeValue, []string{"camera", "type", "object", "detail"}, blueiris.BlueIris, CollectBool{true: []int{2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21}}, "blueIrisServerMetrics"),
+		1:  newMetric("ai_duration", "Duration of Blue Iris AI analysis", prometheus.GaugeValue, []string{"camera", "type", "object", "detail"}, blueiris.BlueIris, CollectBool{true: []int{2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22}}, "blueIrisServerMetrics"),
 		2:  newMetric("ai_count", "Count of Blue Iris AI analysis", prometheus.GaugeValue, []string{"camera", "type"}, blueiris.BlueIris, CollectBool{false: nil}, "blueIrisServerMetrics"),
 		3:  newMetric("ai_duration_distinct", "Duration of Blue Iris AI analysis once", prometheus.GaugeValue, []string{"camera", "type", "object", "detail"}, blueiris.BlueIris, CollectBool{false: nil}, "blueIrisServerMetrics"),
 		4:  newMetric("ai_restarted", "Times BlueIris restarted Deepstack", prometheus.GaugeValue, []string{}, blueiris.BlueIris, CollectBool{false: nil}, "blueIrisServerMetrics"),
@@ -42,6 +42,7 @@ var (
 		19: newMetric("parse_errors_total", "Count of all the errors parsing log lines", prometheus.GaugeValue, []string{}, blueiris.BlueIris, CollectBool{false: nil}, "blueIrisServerMetrics"),
 		20: newMetric("ai_starting", "Count of AI is being started log lines", prometheus.GaugeValue, []string{}, blueiris.BlueIris, CollectBool{false: nil}, "blueIrisServerMetrics"),
 		21: newMetric("ai_started", "Count of AI has been started log lines", prometheus.GaugeValue, []string{}, blueiris.BlueIris, CollectBool{false: nil}, "blueIrisServerMetrics"),
+		22: newMetric("profile", "Count of activation of profiles", prometheus.GaugeValue, []string{"profile"}, blueiris.BlueIris, CollectBool{false: nil}, "blueIrisServerMetrics"),
 	}
 
 	scrapeDurationDesc = prometheus.NewDesc(

